diff --git a/pihole b/pihole
index 8be03f7..8670538 100755
--- a/pihole
+++ b/pihole
@@ -107,7 +107,7 @@ restartDNS() {
   # Determine if we should reload or restart restart
   if [[ "${svcOption}" =~ "reload" ]]; then
     # Using SIGHUP will NOT re-read any *.conf files
-    svc="killall -s SIGHUP ${resolver}"
+    svc="kill -s SIGHUP $(pgrep ${resolver})"
   else
     # Get PID of resolver to determine if it needs to start or restart
     if pidof pihole-FTL &> /dev/null; then
